src/app/services/backend.service.ts

```diff
@@ -99,13 +99,13 @@ export class BackendService {
       })
   }
 
-  getCustomer() {
-    const url = this.baseUrl + `api/customers/2`
+  getCustomer(id) {
+    const url = this.baseUrl + `api/customers/${id}`;
     return this.http.get(url).toPromise()
   }
 
-  editCustomer(data) {
-    const userUrl = this.baseUrl + `api/customers/2/edit`;
+  editCustomer(data, id) {
+    const userUrl = this.baseUrl + `api/customers/${id}/edit`;
```

Comment:
So it looks like the data param here already has an id as a property. Don't pass that id in again as a second parameter, you could just as easily used `${data.id}`